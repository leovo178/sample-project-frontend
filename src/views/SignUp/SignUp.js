import React from 'react'
import { withRouter } from 'react-router-dom'

import { Box } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'

import { useMutation } from '@apollo/react-hooks'

import { FormEditor } from '@views_components'

import { CREATE_USER } from '@views/User/gql/mutation'

const useStyles = makeStyles({
	root: {
		height: '100vh',
		width: '100%',
		display: 'flex',
		alignItems: 'center',
	},
})

const SignUp = props => {
	const { history } = props
	const classes = useStyles()

	const [createNewUser] = useMutation(CREATE_USER)

	const createUser = ({ email, name, password }) => {
		createNewUser({
			variables: { user: { email, name, password } },
		}).then(navigateToSignInPage)
	}

	const navigateToSignInPage = () => history.push('/sign-in')

	return (
		<Box className={classes.root}>
			<FormEditor
				onSubmit={createUser}
				onCancel={() => history.push('/sign-in')}
			/>
		</Box>
	)
}

export default withRouter(SignUp)
