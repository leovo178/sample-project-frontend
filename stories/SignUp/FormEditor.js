import React, { useState } from 'react'
import { Box, Button, TextField, Typography } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles'
import { action } from '@storybook/addon-actions'
import { DeleteDialog } from '@views_components'
import LinkTo from '@storybook/addon-links/react'

const useStyles = makeStyles(theme => ({
	root: {
		width: 381,
		margin: '20px auto',
		padding: theme.spacing(2),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
	},
	formtitle: {
		color: theme.palette.primary.main,
		fontWeight: 600,
	},
	forminput: {
		width: '100%',
		marginTop: 18,
	},
	formcontent: {
		padding: 0,
	},
	formbuttons: {
		display: 'flex',
		flexDirection: 'column',
		padding: 0,
		width: '100%',
		'&>button': {
			marginTop: theme.spacing(2),
			marginLeft: 0,
			'&:not(:first-child)': {
				marginLeft: 0,
			},
		},
	},
	formbutton: {
		color: theme.palette.common.white,
		fontWeight: 600,
		textTransform: 'capitalize',
		padding: '18px 0',
	},
}))

const FormEditor = () => {
	const classes = useStyles()

	const [email, setEmail] = useState('')
	const [name, setName] = useState('')
	const [password, setPassword] = useState('')
	const [confirmPassword, setConfirmPassword] = useState('')
	const [openConfirmDeleteDialog, setOpenConfirmDeleteDialog] = useState(false)
	return (
		<Box className={classes.root}>
			<Typography variant='h5' className={classes.formtitle}>
				Sign up
			</Typography>
			<div className={classes.formcontent}>
				<TextField
					value={email}
					label='EMAIL'
					variant='outlined'
					type='email'
					className={classes.forminput}
					onChange={e => setEmail(e.target.value.toLowerCase())}
				/>
				<TextField
					value={name}
					label='NAME'
					variant='outlined'
					type='text'
					autoComplete='true'
					className={classes.forminput}
					onChange={e => setName(e.target.value)}
				/>
				<TextField
					id='formeditor-password'
					value={password}
					label='PASSWORD'
					variant='outlined'
					type='password'
					autoComplete='true'
					className={classes.forminput}
					onChange={e => setPassword(e.target.value)}
				/>
				<TextField
					id='formeditor-password-confirm'
					value={confirmPassword}
					label='PASSWORD CONFIRM'
					variant='outlined'
					type='password'
					autoComplete='true'
					className={classes.forminput}
					onChange={e => setConfirmPassword(e.target.value)}
				/>
			</div>
			<div className={classes.formbuttons}>
				<Button
					data-cy='submit-button'
					color='primary'
					variant='contained'
					size='large'
					fullWidth
					className={classes.formbutton}
					onClick={() => action('Handle register')({ email, password, name })}
				>
					Register
				</Button>
				<Button
					variant='contained'
					size='large'
					fullWidth
					className={classes.formbutton}
				>
					<LinkTo kind='component-api-signin' story='sign-in'>
						Cancel
					</LinkTo>
				</Button>
			</div>
			<DeleteDialog
				open={openConfirmDeleteDialog}
				onClose={() => {
					setOpenConfirmDeleteDialog(false)
				}}
				// onAgree={onAgreeDeleteAnUser}
				onDisagree={() => {
					setOpenConfirmDeleteDialog(false)
				}}
			/>
		</Box>
	)
}

export default FormEditor
