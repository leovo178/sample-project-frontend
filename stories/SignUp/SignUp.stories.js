import React from 'react'
import SignUp from './SignUp'
import { ThemeProvider } from '@material-ui/core'
import muiTheme from '../../src/theme/muiTheme'

export default {
	title: 'Component Api|SignUp',
	// component: SignUp,
	includeStories: [],
}

export const signUp = () => {
	return (
		<ThemeProvider theme={muiTheme}>
			<SignUp />
		</ThemeProvider>
	)
}
