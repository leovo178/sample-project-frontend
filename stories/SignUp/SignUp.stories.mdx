import { Meta, Story, Preview,Props } from '@storybook/addon-docs/blocks';
import SignUp from './SignUp'

import * as stories from './SignUp.stories.js';

<Meta title="Component Api|SignUp" />

# SignUp
With `MDX` we can define a story for `Sign Up` right in the middle of our
markdown documentation.




  <Story name="SignUp">
    {stories.signUp()}
  </Story>


```jsx 

<Route path='/sign-in' component={SignUp} />
```

# PropTypes

<Props of={SignUp}/>

