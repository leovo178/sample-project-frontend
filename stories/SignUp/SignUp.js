import React from 'react'
import FormEditor from './FormEditor'

const SignUp = () => {
	return <FormEditor />
}

export default SignUp

SignUp.propTypes = {
	// history: PropTypes.object,
}

SignUp.defaultProps = {}
