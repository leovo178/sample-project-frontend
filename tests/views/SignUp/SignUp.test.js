import React from 'react'

import { act } from '@testing-library/react'

import SignUp from '@views/SignUp'

import { findDOMNodeOfSignUp } from './helpers'

describe('<SignUp />', () => {
	let rendered
	beforeEach(async () => {
		await act(async () => {
			rendered = findDOMNodeOfSignUp(<SignUp />)
		})
	})

	it('should match snapshot', async () => {
		expect(rendered.container).toMatchSnapshot()
	})
})
