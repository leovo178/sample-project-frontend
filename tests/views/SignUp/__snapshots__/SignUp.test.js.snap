// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should match snapshot 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-10 makeStyles-root-4"
    >
      <h5
        class="MuiTypography-root makeStyles-formtitle-5 MuiTypography-h5"
        data-testid="formeditor-title"
      >
        Sign up
      </h5>
      <div
        class="makeStyles-formcontent-7"
      >
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-forminput-6"
          data-testid="formeditor-email-input"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            EMAIL
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="true"
              class="MuiInputBase-input MuiOutlinedInput-input"
              placeholder="Email"
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-99 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-101"
              >
                <span>
                  EMAIL
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-forminput-6"
          data-testid="formeditor-name-input"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            NAME
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="true"
              class="MuiInputBase-input MuiOutlinedInput-input"
              placeholder="Name"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-99 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-101"
              >
                <span>
                  NAME
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-forminput-6"
          data-testid="formeditor-password-input"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            PASSWORD
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="true"
              class="MuiInputBase-input MuiOutlinedInput-input"
              placeholder="Password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-99 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-101"
              >
                <span>
                  PASSWORD
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-forminput-6"
          data-testid="formeditor-password-confirm-input"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
          >
            PASSWORD CONFIRM
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              autocomplete="true"
              class="MuiInputBase-input MuiOutlinedInput-input"
              placeholder="Password Confirm"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-99 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-101"
              >
                <span>
                  PASSWORD CONFIRM
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-formbuttons-8"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formbutton-9 MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
          data-testid="formeditor-submit-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Register
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-formbutton-9 MuiButton-containedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
          data-testid="formeditor-cancel-button"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Cancel
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
